lib/composite_primary_keys/composite_predicates.rb

```diff
@@ -51,9 +51,43 @@ def cpk_join_predicate(table1, key1, table2, key2)
     end
 
     def cpk_in_predicate(table, primary_keys, ids)
-      and_predicates = ids.map do |id|
-        cpk_id_predicate(table, primary_keys, id)
+      if primary_keys.length == 2
+        low_cardinality_key_part, high_cardinality_key_part = if ids.map(&:second).uniq.size > ids.map(&:first).uniq.size
```

Comment:
This loops over the ids twice which may impact performance with a lot of ids. Probably better to have a single loop?

Also I wonder if having a hash table keyed on id with the values the count of each item would perform better? Avoids the uniq calls. Would guess that would be better if there are a lot of duplicates and worse if they aren't.\